### ISSUE TITLE
LineEdit placeholder is not aligned correctly with center and right alignment.

### ISSUE THREAD

**Issue by nyenye** [godotengine/godot#18184]
___

***Bugsquad note:** This issue has been confirmed several times already. No need to confirm it further.*

___

**Godot version:**
Godot 3.0.2

**OS/device including version:**
Archlinux

**Issue description:**
Placeholder on LineEdits is misplaced when alignment is center or right. Shown on image below. Username is Left, Password is Center. When alignment is right, it doesn't show (it probably overflows and clips)

![godot_bug](https://user-images.githubusercontent.com/11387253/38754501-658b9868-3f62-11e8-82c6-c940cd5713e1.png)

**Comment by Calinou** [comment]
I can reproduce this on commit https://github.com/godotengine/godot/commit/59b816679e1c87d17ec48bfb5677fb9ff8a9ce5d. Pictured below are labels containing text and placeholder text, with left, center and right alignments:

![LineEdit alignment issue](https://user-images.githubusercontent.com/180032/38894900-ddd2ee64-428e-11e8-8824-d67b7ee516f9.png)

**Comment by Nobelix** [comment]
Can reproduce on 3.0.2 stable
![image](https://user-images.githubusercontent.com/30960698/39405690-96642a56-4ba9-11e8-910b-d7676d4f805a.png)

**Comment by nyenye** [comment]
@akien-mga For me, it's only placeholders, with v3.0.2.stable.custom_build.

Center:
![center](https://user-images.githubusercontent.com/11387253/39512056-d23caf42-4def-11e8-81e5-a30c732746c0.jpg)
![right](https://user-images.githubusercontent.com/11387253/39512060-d502edcc-4def-11e8-8dd4-11def07746b2.jpg)
![left](https://user-images.githubusercontent.com/11387253/39512064-d6e3c436-4def-11e8-9dcb-2657bd7124c8.jpg)

**Comment by robfram** [comment]
It's strange, #18095 should have fixed all of this. Could you try the test project attached in #17113?
I will take a look as soon as I can.

**Comment by Nobelix** [comment]
Maybe Calinou meant can't instead of can

**Comment by nyenye** [comment]
@Nobelix Ooh, indeed could be the case.

@robfram I'll try it out and update this comment.

**Comment by Calinou** [comment]
On my screenshot, all fields have a placeholder defined but only half of them have text entered (so the placeholder displays on the other half instead).

**Comment by robfram** [comment]
Ok, I think I found the problem. It seems to be a problem updating the placeholder position and not calculating it.

TL;DR at end.

Try this steps to confirm all you get the same results as I do:
1. Create a `LineEdit`
2. Set `Text` to anything (i.e. `Testing`) (it shows `Testing` correctly at left position)
3. Change `Alignment` to `Center` (it shows `Testing` correctly at center position)
4. Set `Text` in `Placeholder` to `Placeholder text` (no changes are shown, still `Testing` correctly at center position)
5. Unset `Text` deleting the `Testing` string (it shows `Placeholder text` bad aligned at center, using the center as the starting point for the text)
6. Edit `Text` in `Placeholder` and simply press `Enter` to accept the same `Placeholder text` string as it has (it shows `Placehoder text` correctly centered)

TL;DR:
For all the `LineEdit` you have wrong placed, edit the `Placeholder/Text` property and set it again to force its correct positioning.

Will try to fix this as soon as possible, but feel free to send a PR.

**Comment by leakim34** [comment]
Still got this issue on 3.1.1

**Comment by FeralBytes** [comment]
This is still a problem in Godot 3.2.2-beta4. I request the issue be reopened.

**Comment by kzerot** [comment]
Hi there. 
Still have same problem, 3.2.3
![image](https://user-images.githubusercontent.com/4415061/95048560-b693f680-06f0-11eb-8d9a-56149d909d3e.png)

**Comment by boleanly** [comment]
Reporting same problem with 3.2.3 stable and 3.2.4 rc3. Seems like the cached_placeholder_width is not updated when the project is opened in the editor. If anyone needs a quick fix, setting the placeholder_text variable on ready in GDScript fixes the alignment.

There might be other cases of the cached_width / cached_placeholder_width not updating cause I've also encountered regular text being aligned wrong.

**Comment by nathanfranke** [comment]
Getting this in 3.3.2.stable

![image](https://user-images.githubusercontent.com/14253836/119568288-f2a8c400-bd72-11eb-8643-6036894c36c6.png)

Editing the text makes the issue go away, but closing and re-opening the scene makes it screw up again.

**Comment by HGCollier** [comment]
Currently still present in v3.3.3

### C# Fix
`
public override void _Ready()
{
    GetNode<LineEdit>("LineEditPath").PlaceholderText = "Placeholder";
}
`

### GD Script Fix
`
func _ready():
    get_node("LineEditPath").placeholder_text = "Placeholder"
`

**Comment by rainlizard** [comment]
The alignment is screwing up in the editor too when changing the font of a LineEdit. I think this is the same issue as this thread's. The issue occurs in-game as well even though I'm not specifically loading a font, I'm using a custom Theme which has a custom font.

In editor:
1. Create a long LineEdit (give it 300 pixels width)
2. Set Text to "abcdefghijklmnopqrstuvwxyz" (something long so the offset is easily visible)
3. Set Align to "Right" or "Center"
4. Scroll down to Font (custom_fonts/font) and load a dynamic font that's larger than the default font

The alignment in the editor will now be incorrect. Requiring you to edit the text to re-correct the alignment.

This solves my issue in-game:
```
func _ready():
	text = text
```

**Comment by akien-mga** [comment]
Fixed by #56837 for 3.5 (was fixed for 4.0 a while ago with a refactor).
